src/mito2/src/sst/parquet/page_reader.rs

```diff
@@ -0,0 +1,92 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Parquet page reader.
+
+use std::collections::VecDeque;
+
+use parquet::column::page::{Page, PageMetadata, PageReader};
+use parquet::errors::Result;
+
+/// A reader that reads from cached pages.
+pub(crate) struct CachedPageReader {
+    /// Cached pages.
+    pages: VecDeque<Page>,
+}
+
+impl CachedPageReader {
+    /// Returns a new reader from existing pages.
+    pub(crate) fn new(pages: &[Page]) -> Self {
+        Self {
+            pages: pages.iter().cloned().collect(),
+        }
+    }
+}
+
+impl PageReader for CachedPageReader {
+    fn get_next_page(&mut self) -> Result<Option<Page>> {
```

Comment:
Better to use `pop` instead of `get`, it has side effects. And `get` often means reading data.

```suggestion
    fn pop_next_page(&mut self) -> Result<Option<Page>> {
```

Reply:
This is a method defined in the `PageReader` so we can't change it.